acceptance-tests/src/test/java/io/blueocean/ath/offline/personalization/FavoritesCardsTest.java

```diff
@@ -0,0 +1,177 @@
+
+    @Rule @Inject
+    public GitRepositoryRule git;
+
+    @Inject @BaseUrl
+    String base;
+
+    @Inject
+    ClassicJobApi jobApi;
+
+    @Inject
+    FreestyleJobFactory freestyleFactory;
+
+    @Inject
+    ClassicPipelineFactory pipelineFactory;
+
+    @Inject
+    MultiBranchPipelineFactory multibranchFactory;
+
+    @Inject
+    FavoritesDashboardPage dashboardPage;
+
+    @Inject
+    ActivityPageFactory activityPageFactory;
+
+    private ResourceResolver resources;
+
+    private HttpRequest httpRequest() {
+        return new HttpRequest(base + "/blue/rest");
+    }
+
+    @Before
+    public void setUp() throws IOException {
+        resources = new ResourceResolver(getClass());
+
+        String user = "alice";
+        logger.info(String.format("deleting any existing favorites for %s", user));
+
+        httpRequest()
+            .Delete("/users/{user}/favorites/")
+            .urlPart("user", user)
+            .auth(user, user)
+            .status(204)
+            .as(Void.class);
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        // wipe out all jobs to avoid causing issues w/ SearchTest
+        jobApi.deleteFolder(FOLDER);
+    }
+
+    @Test
+    public void testFreestyle() throws IOException {
+        String jobName = "favoritescards-freestyle";
+        FreestyleJob freestyle = freestyleFactory.pipeline(FOLDER, jobName).create("echo hello\nsleep 5\necho world");
+        String fullName = freestyle.getFullName();
+
+        dashboardPage.open();
+        dashboardPage.togglePipelineListFavorite(jobName);
+        dashboardPage.checkFavoriteCardCount(1);
+        dashboardPage.clickFavoriteCardRunButton(fullName);
+        dashboardPage.checkFavoriteCardStatus(fullName, RUNNING, SUCCESS);
+        dashboardPage.removeFavoriteCard(fullName);
+        dashboardPage.checkFavoriteCardCount(0);
+        dashboardPage.checkIsPipelineListItemFavorited(jobName, false);
+    }
+
+    @Test
+    public void testClassicPipeline() throws IOException {
+        String jobName = "favoritescards-pipeline";
+        String script = resources.loadJenkinsFile();
+        ClassicPipeline pipeline = pipelineFactory.pipeline(FOLDER, jobName).createPipeline(script).build();
+        String fullName = pipeline.getFullName();
+
+        dashboardPage.open();
+        dashboardPage.togglePipelineListFavorite(jobName);
+        dashboardPage.checkFavoriteCardCount(1);
+
+        dashboardPage.checkFavoriteCardStatus(fullName, SUCCESS);
+        dashboardPage.clickFavoriteCardRunButton(fullName);
+        dashboardPage.checkFavoriteCardStatus(fullName, RUNNING, SUCCESS);
+        dashboardPage.clickFavoriteCardReplayButton(fullName);
+        dashboardPage.checkFavoriteCardStatus(fullName, RUNNING, SUCCESS);
+        dashboardPage.removeFavoriteCard(fullName);
+        dashboardPage.checkFavoriteCardCount(0);
+        dashboardPage.checkIsPipelineListItemFavorited(jobName, false);
```

Comment:
I want to do a little more refactoring to the code so that page objects use builder pattern so we can chain methods together and eliminate a little boilerplate.

Reply:
Going to defer these changes to a future PR.